CRM/Contribute/Form/Contribution.php

```diff
@@ -1585,9 +1585,6 @@ protected function submit($submittedValues, $action, $pledgePaymentID) {
         $params['participant_id'] = $pId;
```

Comment:
@eileenmcnaughton why don't we also get rid of this assignment
```diff
- $params['contribution_mode'] = 'participant';
```

And later in Contribution::create
```diff
-  if (CRM_Utils_Array::value('contribution_mode', $params) == 'participant') {
+ if (!empty($params['participant_id'])) {
      $entityId = $params['participant_id'];
      $entityTable = 'civicrm_participant';
      $additionalParticipantId = CRM_Event_BAO_Participant::getAdditionalParticipantIds($entityId);
    }
    elseif (!empty($params['membership_id'])) {
      //so far $params['membership_id'] should only be set coming in from membershipBAO::create so the situation where multiple memberships
      // are created off one contribution should be handled elsewhere
      $entityId = $params['membership_id'];
      $entityTable = 'civicrm_membership';
    }
    else {
      $entityId = $params['contribution']->id;
      $entityTable = 'civicrm_contribution';
    }
    ```

Reply:
@monishdeb I'm happy to - I didn't remove it because I haven't worked through anything on the participant side yet & am only focussing on membership so far - I was going to switch my focus once the current PRs are merged

Reply:
Acutally that  is not quite true - I had started on the participant side over here https://github.com/civicrm/civicrm-core/pull/20650 & I wanted to get that additional test cover before going further on it